sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -131,6 +131,26 @@ case class CreateViewCommand(
         s"specified by CREATE VIEW (num: `${userSpecifiedColumns.length}`).")
     }
 
+    // When creating a permanent view, not allowed to reference temporary objects.
+    if (!isTemporary) {
```

Comment:
also i'd move the check into a function, so it is more obvious what's going on with the main workflow.